src/Orleans/Logging/TraceLogger.cs

```diff
@@ -1,1222 +1,13 @@
-        {
-            foreach (var  tc in TelemetryConsumers.OfType<IMetricTelemetryConsumer>())
-            {
-                tc.DecrementMetric(name, value);
-            }
-        }
-
-        public override void TrackRequest(string name, DateTimeOffset startTime, TimeSpan duration, string responseCode, bool success)
-        {
-            foreach (var tc in TelemetryConsumers.OfType<IRequestTelemetryConsumer>())
-            {
-                tc.TrackRequest(name, startTime, duration, responseCode, success);
-            }
-        }
-
-        public override void TrackException(Exception exception, IDictionary<string, string> properties = null, IDictionary<string, double> metrics = null)
-        {
-            foreach (var tc in TelemetryConsumers.OfType<IExceptionTelemetryConsumer>())
-            {
-                tc.TrackException(exception, properties, metrics);
-            }
-        }
-
-        public override void TrackTrace(string message)
-        {
-            foreach (var tc in TelemetryConsumers.OfType<ITraceTelemetryConsumer>())
-            {
-                tc.TrackTrace(message);
-            }
-        }
-
-        public override void TrackTrace(string message, Severity severity)
-        {
-            foreach (var tc in TelemetryConsumers.OfType<ITraceTelemetryConsumer>())
-            {
-                tc.TrackTrace(message, severity);
-            }
-        }
-
-        public override void TrackTrace(string message, Severity severity, IDictionary<string, string> properties)
-        {
-            foreach (var tc in TelemetryConsumers.OfType<ITraceTelemetryConsumer>())
-            {
-                tc.TrackTrace(message, severity, properties);
-            }
-        }
-
-        public override void TrackTrace(string message, IDictionary<string, string> properties)
-        {
-            foreach (var tc in TelemetryConsumers.OfType<ITraceTelemetryConsumer>())
-            {
-                tc.TrackTrace(message, properties);
-            }
-        }
-
-        #endregion
-
-        /// <summary>
-        /// This custom comparer lets us sort the TraceLevelOverrides list so that the longest prefix comes first
-        /// </summary>
-        private class TraceOverrideComparer : Comparer<Tuple<string, Severity>>
-        {
-            public override int Compare(Tuple<string, Severity> x, Tuple<string, Severity> y)
-            {
-                return y.Item1.Length.CompareTo(x.Item1.Length);
-            }
-        }
-
-        private static class NativeMethods
-        {
-            [DllImport("Dbghelp.dll")]
-            public static extern bool MiniDumpWriteDump(
-                IntPtr hProcess,
-                int processId,
-                IntPtr hFile,
-                MiniDumpType dumpType,
-                IntPtr exceptionParam,
-                IntPtr userStreamParam,
-                IntPtr callbackParam);
-        }
-    }
-
-    internal enum MiniDumpType
+    [Obsolete]
```

Comment:
Obsolete can take a message that will show up in the error list. Please use this to explain what's the alternative they should be using (Logger for producing new events, LogFormatter for formatting when producing and LogManager for configuration of the sinks)